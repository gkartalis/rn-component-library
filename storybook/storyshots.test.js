import initStoryShots from '@storybook/addon-storyshots';

/*
  This will automatically take all stories and
  generate snapshot tests for all your stories
  for free inside storybook/__snapshots__
*/

initStoryShots();
