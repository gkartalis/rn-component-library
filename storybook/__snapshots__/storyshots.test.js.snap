// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button default 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#202c41",
          "borderColor": "#202c41",
          "borderRadius": 6,
          "borderWidth": 1,
          "marginVertical": 7,
          "opacity": 1,
          "paddingVertical": 14,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "500",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Button outline 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#c6c6c6",
          "borderRadius": 6,
          "borderWidth": 1,
          "marginVertical": 7,
          "opacity": 1,
          "paddingVertical": 14,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "fontSize": 18,
              "fontWeight": "500",
            },
            Object {
              "color": "#202c41",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form default 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#e6e6e6",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form with header 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 28,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#202c41",
              "fontSize": 32,
              "fontWeight": "600",
              "marginBottom": 12,
            }
          }
        >
          Hello
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#e6e6e6",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form with header and subheader 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 28,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#202c41",
              "fontSize": 32,
              "fontWeight": "600",
              "marginBottom": 12,
            }
          }
        >
          Hello
        </Text>
        <Text
          style={
            Object {
              "color": "#9ca5ab",
              "fontSize": 20,
              "marginBottom": 12,
            }
          }
        >
          Welcome back. Kindly enter your login details.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#e6e6e6",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/FieldWrapper default 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/FieldWrapper error message 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#b55464",
            },
          ]
        }
      >
        Please enter a valid Email
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/FieldWrapper multiple fields 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#b55464",
            },
          ]
        }
      >
        Please enter a valid Email
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#50c356",
            },
          ]
        }
      >
        Looks legit!
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/FieldWrapper success message 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <Text>
        Hello, wrapper.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#50c356",
            },
          ]
        }
      >
        Looks legit!
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/Switch default 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Agree to Terms
      </Text>
      <RCTSwitch
        accessibilityRole="button"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={true}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/Switch with error 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Agree to Terms
      </Text>
      <RCTSwitch
        accessibilityRole="button"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 31,
            "width": 51,
          }
        }
        value={false}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#b55464",
            },
          ]
        }
      >
        You must agree to the Terms
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput default 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Demo
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput email 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Email
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
        value="kartalisge@gmail.com"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput password 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Password
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
        value="Password"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput with error message 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Demo
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
            Object {
              "color": "#b55464",
            },
          ]
        }
      >
        This is an error
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput with placeholder 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Demo
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="demo placeholder"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Storyshots Form/TextInput with value 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 40,
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9ca5ab",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Demo
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "500",
            "paddingBottom": 10,
          }
        }
        underlineColorAndroid="transparent"
        value="hello value"
      />
      <View
        style={
          Object {
            "backgroundColor": "#c6c6c6",
            "height": 1,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#9ca5ab",
              "fontSize": 13,
              "marginTop": 5,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
